src/tools/tidy/src/deps.rs

```diff
@@ -501,27 +504,7 @@ fn check_permitted_dependencies(
     let mut deps = HashSet::new();
     for to_check in restricted_dependency_crates {
         let to_check = pkg_from_name(metadata, to_check);
-        use cargo_platform::Cfg;
-        use std::str::FromStr;
-        // We don't expect the compiler to ever run on wasm32, so strip
-        // out those dependencies to avoid polluting the permitted list.
-        deps_of_filtered(metadata, &to_check.id, &mut deps, &|dep_kinds| {
-            dep_kinds.iter().any(|dep_kind| {
-                dep_kind
-                    .target
-                    .as_ref()
-                    .map(|target| {
-                        !target.matches(
-                            "wasm32-unknown-unknown",
-                            &[
-                                Cfg::from_str("target_arch=\"wasm32\"").unwrap(),
-                                Cfg::from_str("target_os=\"unknown\"").unwrap(),
-                            ],
-                        )
```

Comment:
I don't know why, but one of suspects is https://docs.rs/cargo-platform/0.1.2/cargo_platform/enum.Platform.html#method.matches

It says "Returns whether the Platform matches the given target and cfg.", but what means "matches"?

Lets skip target part and look at cfg.

There can be 2 situations:
1. `cfg(not(windows))` matches `cfg(target_arch="wasm32")` (matches currently): which can be understand as: if X *can be* Y? But `cfg(not(windows))` for example can be `x86_64-unknown-linux-gnu` which actually not matches `cfg(target_arch="wasm32")`.
2. If X has cfgs which *is* Y? `cfg(all(target_arch="wasm32",target_os="unknown"))` matches `cfg(target_arch="wasm32")` because `target_arch` is "wasm32".

But 2. can be broken again if `cfg(any(target_arch="wasm32", target_arch="x86_64"))` matches against `cfg(target_arch="wasm32")`, as it can be `x86_64-unknown-linux-gnu` again, which is not `cfg(target_arch="wasm32")`.

Reply:
Okay, looks like this don't work as i expected:
`Name("wasm32-wasi").matches("", &[Cfg::from_str("target_arch=\"wasm32\"").unwrap(),]) == false`, while should return true (ignore part that Platform::matches checks target *or* cfg).
`cargo_platform` works on strings and don't know to parse triples or if one cfg turn on other cfgs.

Reply:
Comments are messy (for me too, in month later), so, please ask question questions.

Reply:
We should try to make this work instead of getting rid of it entirely. If we really want to remove it, we can revert the changes that were made, but I don't believe this is the best solution for this situation.